File: .NET/eStationDemo/Entity/TagResult.cs
using MessagePack;

namespace eStationDemo.Model
{
    [MessagePackObject]
    /// <summary>
    /// Tag result
    /// </summary>
    public class TagResult
    {
        /// <summary>
        /// Tag ID
        /// </summary>
        [Key(0)]
        public string TagID { get; set; } = string.Empty;
        /// <summary>
        /// RF power
        /// </summary>
        [Key(1)]
        public int RfPower { get; set; } = -256;
        /// <summary>
        /// Battery
        /// </summary>
        [Key(2)]
        public byte Battery { get; set; } = 0;
        /// <summary>
        /// Screen
        /// </summary>
        [Key(3)]
        public byte Version { get; set; } = 0;
        /// <summary>
        /// Version
        /// </summary>
        [Key(4)]
        public byte Status { get; set; } = 0;
        /// <summary>
        /// Token
        /// </summary>
        [Key(5)]
        public int Token { get; set; } = 0;
        /// <summary>
        /// Temperature
        /// </summary>
        [Key(6)]
        public int Temperature { get; set; } = 0;
        /// <summary>
        /// Channel
        /// </summary>
        [Key(7)]
        public int Channel { get; set; } = 0;
        /// <summary>
        /// [DSL]UTC time, 
        /// </summary>
        [Key(8)]
        public byte[] UtcTime { get; set; } = [];
        /// <summary>
        /// [DSL]UTC time, 
        /// </summary>
        [Key(9)]
        public byte[] TimePercent { get; set; } = [];
        /// <summary>
        /// [DSL]Count
        /// </summary>
        [Key(10)]
        public byte Count { get; set; } = 0;
    }
}

File: Python/Entities/tagResult.py
import msgpack
from dataclasses import dataclass, field
from typing import List

@dataclass
class TagResult:
    """
    Tag result
    """
    TagId: str = ""
    RfPower: int = -256
    Battery: int = 0
    Version: int = 0
    Status: int = 0
    Token: int = 0
    Temperature: int = 0
    Channel: int = 0
    UtcTime: bytes = field(default_factory=bytes)
    TimePercent: bytes = field(default_factory=bytes)
    Count: int = 0

    def to_msgpack(self) -> bytes:
        data = [
            self.TagId,
            self.RfPower,
            self.Battery,
            self.Version,
            self.Status,
            self.Token,
            self.Temperature,
            self.Channel,
            self.UtcTime,
            self.TimePercent,
            self.Count
        ]
        return msgpack.packb(data)

    @staticmethod
    def from_msgpack(packed: bytes):
        data = msgpack.unpackb(packed)
        return TagResult(
            TagId=data[0],
            RfPower=data[1],
            Battery=data[2],
            Version=data[3],
            Status=data[4],
            Token=data[5],
            Temperature=data[6],
            Channel=data[7],
            UtcTime=data[8],
            TimePercent=data[9],
            Count=data[10]
        )